File: ios/ReservationApp/ReservationApp/Presentation/Views/Admin/Dashboard/TodayReservationsView.swift
//
//  TodayReservationsView.swift
//  ReservationApp
//
//  Created by Mehmet Ali Kısacık on 13.11.2025.
//

import SwiftUI

struct TodayReservationsView: View {
    let reservations: [TodayReservationGroup]
    let isLoading: Bool
    
    var body: some View {
        VStack(alignment: .leading, spacing: ThemeManager.shared.spacing.md) {
            Text("Bugünkü Rezervasyonlar")
                .font(AppTypography.h6())
                .fontWeight(.semibold)
                .foregroundColor(AppColors.textPrimary)
            
            if isLoading {
                ProgressView()
                    .frame(maxWidth: .infinity)
                    .padding(ThemeManager.shared.spacing.xl)
            } else if reservations.isEmpty {
                Text("Bugün için rezervasyon bulunmamaktadır.")
                    .font(AppTypography.body2())
                    .foregroundColor(AppColors.textSecondary)
                    .frame(maxWidth: .infinity)
                    .padding(ThemeManager.shared.spacing.xl)
            } else {
                ForEach(reservations) { reservation in
                    TodayReservationRow(reservation: reservation)
                }
            }
        }
        .padding(ThemeManager.shared.spacing.lg)
        .background(AppColors.backgroundPaper)
        .cornerRadius(ThemeManager.shared.borderRadius.card.value)
        .shadow(color: Color.black.opacity(0.05), radius: 4, x: 0, y: 2)
    }
}

struct TodayReservationRow: View {
    let reservation: TodayReservationGroup
    
    var body: some View {
        HStack {
            VStack(alignment: .leading, spacing: ThemeManager.shared.spacing.xs) {
                Text(getTurkishName(reservation.mealTimeSlotName))
                    .font(AppTypography.body1())
                    .fontWeight(.medium)
                    .foregroundColor(AppColors.textPrimary)
                
                Text("\(formatTimeRange(reservation.startTime, reservation.endTime)) • \(reservation.restaurantName)")
                    .font(AppTypography.body2())
                    .foregroundColor(AppColors.textSecondary)
            }
            
            Spacer()
            
            Text("\(reservation.reservationCount)")
                .font(AppTypography.h6())
                .fontWeight(.semibold)
                .foregroundColor(AppColors.primaryLight)
        }
        .padding(.vertical, ThemeManager.shared.spacing.sm)
    }
    
    private func getTurkishName(_ englishName: String) -> String {
        switch englishName.lowercased() {
        case "breakfast":
            return "Kahvaltı"
        case "lunch":
            return "Öğle Yemeği"
        case "dinner":
            return "Akşam Yemeği"
        default:
            return englishName
        }
    }
    
    private func formatTimeRange(_ start: String, _ end: String) -> String {
        // Format time range - assuming times are in HH:mm format
        // If backend returns full ISO strings, we'll need to parse them
        if start.contains("T") {
            // ISO format - extract time part
            let startTime = String(start.split(separator: "T").last?.prefix(5) ?? "")
            let endTime = String(end.split(separator: "T").last?.prefix(5) ?? "")
            return "\(startTime)-\(endTime)"
        } else {
            return "\(start)-\(end)"
        }
    }
}

File: android/app/src/main/java/com/reservationapp/ui/admin/components/TodayReservationsView.kt
package com.reservationapp.ui.admin.components

import androidx.compose.foundation.layout.*
import androidx.compose.material3.Card
import androidx.compose.material3.CardDefaults
import androidx.compose.material3.CircularProgressIndicator
import androidx.compose.material3.MaterialTheme
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.dp
import com.reservationapp.core.ui.theme.BackgroundPaper
import com.reservationapp.core.ui.theme.PrimaryLight
import com.reservationapp.core.ui.theme.Spacing
import com.reservationapp.core.ui.theme.TextPrimary
import com.reservationapp.core.ui.theme.TextSecondary
import com.reservationapp.domain.model.TodayReservationGroup

@Composable
fun TodayReservationsView(
    reservations: List<TodayReservationGroup>,
    isLoading: Boolean,
    modifier: Modifier = Modifier
) {
    Card(
        modifier = modifier.fillMaxWidth(),
        colors = CardDefaults.cardColors(
            containerColor = BackgroundPaper
        ),
        elevation = CardDefaults.cardElevation(defaultElevation = 2.dp)
    ) {
        Column(
            modifier = Modifier
                .fillMaxWidth()
                .padding(Spacing.lg),
            verticalArrangement = Arrangement.spacedBy(Spacing.lg)
        ) {
            Text(
                text = "Bugünkü Rezervasyonlar",
                style = MaterialTheme.typography.titleMedium.copy(
                    fontWeight = FontWeight.SemiBold
                ),
                color = TextPrimary
            )

            if (isLoading) {
                Box(
                    modifier = Modifier
                        .fillMaxWidth()
                        .padding(Spacing.xl),
                    contentAlignment = Alignment.Center
                ) {
                    CircularProgressIndicator()
                }
            } else if (reservations.isEmpty()) {
                Text(
                    text = "Bugün için rezervasyon bulunmamaktadır.",
                    style = MaterialTheme.typography.bodyMedium,
                    color = TextSecondary,
                    modifier = Modifier
                        .fillMaxWidth()
                        .padding(Spacing.xl)
                )
            } else {
                reservations.forEach { reservation ->
                    TodayReservationRow(reservation = reservation)
                }
            }
        }
    }
}

@Composable
private fun TodayReservationRow(
    reservation: TodayReservationGroup
) {
        Row(
        modifier = Modifier
            .fillMaxWidth()
            .padding(vertical = Spacing.sm),
        horizontalArrangement = Arrangement.SpaceBetween,
        verticalAlignment = Alignment.CenterVertically
    ) {
        Column(
            modifier = Modifier.weight(1f),
            verticalArrangement = Arrangement.spacedBy(Spacing.xs)
        ) {
            Text(
                text = getTurkishMealTimeSlotName(reservation.mealTimeSlotName),
                style = MaterialTheme.typography.bodyLarge.copy(
                    fontWeight = FontWeight.Medium
                ),
                color = TextPrimary
            )
            Text(
                text = "${formatTimeRange(reservation.startTime, reservation.endTime)} • ${reservation.restaurantName}",
                style = MaterialTheme.typography.bodyMedium,
                color = TextSecondary
            )
        }
        Text(
            text = "${reservation.reservationCount}",
            style = MaterialTheme.typography.titleMedium.copy(
                fontWeight = FontWeight.SemiBold
            ),
            color = PrimaryLight
        )
    }
}

private fun getTurkishMealTimeSlotName(englishName: String): String {
    return when (englishName.lowercase()) {
        "breakfast" -> "Kahvaltı"
        "lunch" -> "Öğle Yemeği"
        "dinner" -> "Akşam Yemeği"
        else -> englishName
    }
}

private fun formatTimeRange(start: String, end: String): String {
    // Handle ISO format or simple time format
    val startTime = if (start.contains("T")) {
        start.split("T").lastOrNull()?.take(5) ?: start
    } else {
        start
    }
    val endTime = if (end.contains("T")) {
        end.split("T").lastOrNull()?.take(5) ?: end
    } else {
        end
    }
    return "$startTime-$endTime"
}

